src/omt/integer_optimizer.cpp

```diff
@@ -0,0 +1,107 @@
+/*********************                                                        */
+/*! \file integer_optimizer.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Yancheng Ou
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Optimizer for Integer type
+ **/
+
+#include "integer_optimizer.h"
+
+#include "options/smt_options.h"
+#include "smt/smt_engine.h"
+#include "theory/quantifiers/quantifiers_attributes.h"
+#include "theory/smt_engine_subsolver.h"
+
+using namespace CVC4::theory;
+
+namespace CVC4::smt {
+
+std::unique_ptr<SmtEngine> OMTOptimizerInteger::createOptChecker(
```

Comment:
It is a bit hard to keep track of all these variants.

It might be better to not have this method and just use the `createOptCheckerWithTimeout` utility directly.

Reply:
Agreed